x/metadata/types/signer_utils.go

```diff
@@ -0,0 +1,429 @@
+package types
+
+import (
+	"bytes"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+)
+
+// PartyDetails is a struct used to help process party and signer validation.
+// Even though all the fields are public, you should usually use the Get/Set methods
```

Comment:
I think this comment might be out of date. It doesn't look like these fields are public anymore.

Reply:
Good call. I'll remove the last 2 comment lines in the next push.